main.tf

```diff
@@ -3,7 +3,7 @@ locals {
   plan_enabled     = local.enabled && var.plan_enabled
   iam_role_enabled = local.enabled && var.iam_role_enabled
   iam_role_name    = coalesce(var.iam_role_name, module.label_backup_role.id)
-  iam_role_arn     = join("", var.iam_role_enabled ? aws_iam_role.default.*.arn : data.aws_iam_role.existing.*.arn)
+  iam_role_arn     = var.iam_role_enabled ? join("", aws_iam_role.default.*.arn) : (var.plan_enabled ? join("", data.aws_iam_role.existing.*.arn) : null)
```

Comment:
So if the role is enabled, it will use the created role. if the plan is enabled, then use the existing role? And if neither are enabled then set to null?

Reply:
yes that is correct